### ISSUE TITLE
no-array-prototype-extensions false positive Iterator.toArray()

### ISSUE THREAD

**Issue by sukima** [ember-cli/eslint-plugin-ember#2362]
```typescript
function* test(): Generator<string> {
  yield 'foo';
  yield 'bar';
  yield 'baz';
  yield 'frotz';
}

test().take(2).toArray(); // => ["foo", "bar"]
// Don't use Ember's array prototype extensions
```
